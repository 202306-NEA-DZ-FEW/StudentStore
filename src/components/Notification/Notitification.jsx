import React from "react";
import { AiFillCloseCircle } from "react-icons/ai";

const Notification = ({ type, message, onClose }) => {
    const backgroundColor = type === "success" ? "#4CAF50" : "#FF5733";

    return (
        <div
            style={{
                backgroundColor,
                color: "white",
                padding: "10px",
                borderRadius: "5px",
                marginBottom: "10px",
            }}
        >
            <p>
                {message}{" "}
                <button onClick={onClose}>
                    <AiFillCloseCircle />
                </button>
            </p>
        </div>
    );
};

export default Notification;
