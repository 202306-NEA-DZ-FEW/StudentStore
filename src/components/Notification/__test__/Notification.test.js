import renderer from "react-test-renderer";
import Notification from "../Notitification";

jest.mock("react-icons/ai", () => ({
    AiFillCloseCircle: "mocked-AiFillCloseCircle",
}));

it("renders correctly", () => {
    const tree = renderer.create(<Notification />).toJSON();
    expect(tree).toMatchSnapshot();
});
