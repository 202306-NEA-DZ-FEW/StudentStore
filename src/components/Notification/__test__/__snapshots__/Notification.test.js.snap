// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#FF5733",
      "borderRadius": "5px",
      "color": "white",
      "marginBottom": "10px",
      "padding": "10px",
    }
  }
>
  <p>
     
    <button>
      <mocked-AiFillCloseCircle />
    </button>
  </p>
</div>
`;
